features/base/test/test_kernel_config.py

```diff
@@ -1,10 +1,25 @@
-from helper.exception import NotPartOfFeatureError, DisabledBy
-from helper.tests.kernel_config import KernelConfig
+from helper.tests.file_content import file_content
 import pytest
 
-def test_kernel_config(client, features):
-    """The test function executed by pytest"""
-    try:
-        KernelConfig(client, features)
-    except (NotPartOfFeatureError, DisabledBy) as e:
-        pytest.skip(str(e))
\ No newline at end of file
+@pytest.mark.parametrize(
+    "args",
+    [
+        (   {
+            # example expected kernel config options:
+            #"CONFIG_INIT_ENV_ARG_LIMIT": "32",
+            #"CONFIG_COMPILE_TEST": "is not set",
+            #"CONFIG_WERROR": "n",
+            #"CONFIG_LOCALVERSION": "",
+            #"CONFIG_LOCALVERSION_AUTO": "is not set",
+            #"CONFIG_HAVE_KERNEL_GZIP": "y"
+            }
+        )
+    ]
+)
+
+def test_kernel_config(client, args):
+    """compare kernel config options from the parametrize section with the
+    kernel config options in the /boot/config-* file. The values 'is not set'
+    and 'n' are treated as the same."""
+    file = "/boot/config-*"
```

Comment:
Instead of `/boot/config-*` I'd rather take an explicit config file. This could be done by evaluating the boot config in real time. However, this wouldn't work on `chroot` platform. Another solution would be to determinate which config is present in this directory. But I think we may skip this in this test since only freshly generated images will be tested. But this could become a problem by an adjusted image that could be tested by `Pytest`.